analyses/cell-type-ewings/aucell-annotation.sh

```diff
@@ -0,0 +1,95 @@
+#!/bin/bash
+
+# This script identfies tumor cells using AUCell for all samples in SCPCP000015
+# Usage: ./aucell-annotation.sh
+
+set -euo pipefail
+
+# navigate to where script lives
+cd $(dirname "$0")
+module_dir=$(pwd)
+
+# use 4 CPUs
+threads=4
+
+# set up input and output file paths
+# make sure full paths are provided for notebook rendering
+data_dir="${module_dir}/../../data/current/SCPCP000015"
+results_dir="${module_dir}/results/aucell_annotation"
+mkdir -p ${results_dir}
+
+# define path to marker genes file
+marker_genes_file="${module_dir}/references/tumor-marker-genes.tsv"
+
+# define scripts and notebook directories
+scripts_dir="scripts/auc-workflow"
+notebook_dir="template_notebooks/auc-workflow"
+
+# define reference sample inputs and outputs
+ref_sample="SCPCS000490"
+ref_library="SCPCL000822"
+ref_sce="${data_dir}/${ref_sample}/${ref_library}_processed.rds"
+
+ref_results_dir="${results_dir}/${ref_sample}"
+mkdir -p $ref_results_dir
+
+ref_auc_results="${results_dir}/${ref_sample}/${ref_library}_auc-classifications.tsv"
+
+echo "Running AUCell for reference sample: $ref_sample and library: $ref_library"
+# first run AUCell on reference sample
+auc_threshold=$(Rscript $scripts_dir/01-run-aucell.R \
+    --sce_file $ref_sce \
+    --output_file $ref_auc_results \
+    --return_auc)
+
+# now run AUCell, gene set scores, and generate AUCell report for each sample that is not the ref sample
+for sample_id in $(basename ${data_dir}/SCPCS*); do
```

Comment:
This doesn't work, but it looks like it did, because apparently the failure is masked from error handling due to being in a loop definition. The issue is that `basename` only takes one argument unless you add the `-a` flag. So I am suggesting the fix, but breaking it onto a second line that would hopefully trigger an error if it fails.
 
```suggestion
sample_ids= $(basename -a ${data_dir}/SCPCS*)
for sample_id in $sample_ids; do
```

Reply:
What do you mean it doesn't work? It looks like it lists all samples to me? 
```
(base) allyhawkins@allys-work-mac OpenScPCA-analysis % basename data/current/SCPCP000015/SCPCS*
SCPCS000490
SCPCS000491
SCPCS000492
SCPCS000493
SCPCS000494
SCPCS000495
SCPCS000496
SCPCS000749
SCPCS000750
SCPCS000751
SCPCS000752
SCPCS000753
SCPCS000754
```

Reply:
It works on a Mac, but not on linux, which has a different `basename` version. See https://github.com/AlexsLemonade/OpenScPCA-analysis/actions/runs/9895475872/job/27335462108#step:6:3131 for the error